sdk/core/azure-core/src/test/java/com/azure/core/ConfigurationTests.java

```diff
@@ -90,28 +89,28 @@ public void foundConfigurationIsConverted() {
      */
     @Test
     public void notFoundConfigurationIsConvertedToNull() {
-        assertNull(ConfigurationManager.getConfiguration().get("invalidConfiguration", String::toUpperCase));
+        assertNull(Configuration.getGlobalConfiguration().get("invalidConfiguration", String::toUpperCase));
     }
 
     @Test
     public void logLevelUpdatesInstantly() {
-        String initialLogLevel = ConfigurationManager.getConfiguration().get(BaseConfigurations.AZURE_LOG_LEVEL);
-        System.setProperty(BaseConfigurations.AZURE_LOG_LEVEL, "123456789");
-        assertNotEquals(initialLogLevel, ConfigurationManager.getConfiguration().get(BaseConfigurations.AZURE_LOG_LEVEL));
+        String initialLogLevel = Configuration.getGlobalConfiguration().get(Configuration.PROPERTY_AZURE_LOG_LEVEL);
+        System.setProperty(Configuration.PROPERTY_AZURE_LOG_LEVEL, "123456789");
+        assertNotEquals(initialLogLevel, Configuration.getGlobalConfiguration().get(Configuration.PROPERTY_AZURE_LOG_LEVEL));
 
         // Cleanup the test
         if (initialLogLevel != null) {
-            System.setProperty(BaseConfigurations.AZURE_LOG_LEVEL, initialLogLevel);
+            System.setProperty(Configuration.PROPERTY_AZURE_LOG_LEVEL, initialLogLevel);
         }
     }
 
     @Test
     public void tracingDisabledUpdatesInstantly() {
-        boolean initialTracingDisabled = Boolean.parseBoolean(ConfigurationManager.getConfiguration().get(BaseConfigurations.AZURE_TRACING_DISABLED));
-        System.setProperty(BaseConfigurations.AZURE_TRACING_DISABLED, Boolean.toString(!initialTracingDisabled));
-        assertNotEquals(initialTracingDisabled, ConfigurationManager.getConfiguration().get(BaseConfigurations.AZURE_TRACING_DISABLED));
+        boolean initialTracingDisabled = Boolean.parseBoolean(Configuration.getGlobalConfiguration().get(Configuration.PROPERTY_AZURE_TRACING_DISABLED));
+        System.setProperty(Configuration.PROPERTY_AZURE_TRACING_DISABLED, Boolean.toString(!initialTracingDisabled));
+        assertNotEquals(initialTracingDisabled, Configuration.getGlobalConfiguration().get(Configuration.PROPERTY_AZURE_TRACING_DISABLED));
 
         // Cleanup the test
-        System.setProperty(BaseConfigurations.AZURE_TRACING_DISABLED, Boolean.toString(initialTracingDisabled));
+        System.setProperty(Configuration.PROPERTY_AZURE_TRACING_DISABLED, Boolean.toString(initialTracingDisabled));
```

Comment:
I would wrap this in a `try/finally` block, if the test fails at line 111, this will never get reset. Same with line 103.

Reply:
Tests never fail! :-)

Reply:
![giphy](https://user-images.githubusercontent.com/10136526/65187328-81066780-da20-11e9-8644-babc56d1e86c.gif)